cNVMe/Controller.cpp

```diff
@@ -357,8 +360,17 @@ namespace cnvme
 				}
 				else if (command.DW10_Identify.CNS == constants::commands::identify::cns::NAMESPACE_ACTIVE) // Identify Namespace
 				{
-					//todo.. actually do something here
-					memset(transferPayload.getBuffer(), 0, transferPayload.getSize()); // Technically, this is ok if we don't have a namespace with the given NSID. 
+					auto namespaceSelected = this->NamespaceIdToNamespace.find(command.NSID);
+					if (namespaceSelected != this->NamespaceIdToNamespace.end())
+					{
+						LOG_INFO("Grabbing Identify Namespace for NSID " + std::to_string(namespaceSelected->first));
+						auto identifyNamespaceStructure = namespaceSelected->second.getIdentifyNamespaceStructure();
+						memcpy_s(transferPayload.getBuffer(), transferPayload.getSize(), &identifyNamespaceStructure, sizeof(identifyNamespaceStructure));
+					}
+					else
+					{
+						memset(transferPayload.getBuffer(), 0, transferPayload.getSize()); // Technically, this is ok if we don't have a namespace with the given NSID
```

Comment:
Add comment in this region describing what causes you to get in this else block and where in the spec it states that you populate the output payload with zeroes.

Reply:
Figure 106 of NVMe 1.3. Added a footnote.